networking/src/NTP/Client.hs

```diff
@@ -1,47 +1,46 @@
 {-# LANGUAGE ConstraintKinds     #-}
 {-# LANGUAGE FlexibleContexts    #-}
 {-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE NoImplicitPrelude   #-}
 {-# LANGUAGE RankNTypes          #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 
 -- | This module implements functionality of NTP client.
 
 module NTP.Client
-    ( startNtpClient
+    ( NtpMonad
+    , withNtpClient
     , NtpClientSettings (..)
     , NtpStopButton (..)
     , ntpSingleShot
     , hoistNtpClientSettings
     ) where
 
-import           Control.Concurrent.STM (atomically, modifyTVar')
-import           Control.Concurrent.STM.TVar (TVar, newTVarIO, readTVar, readTVarIO, writeTVar)
-import           Control.Exception.Safe (Exception, MonadMask, bracketOnError, catchAny, handleAny,
-                                         throwM)
+import           Universum hiding (log)
+
+import           Control.Concurrent.STM (modifyTVar')
+import           Control.Concurrent.STM.TVar (TVar, readTVar, writeTVar)
+import           Control.Exception.Safe (Exception, MonadMask, bracketOnError, catchAny, handleAny)
```

Comment:
I didn't notice how we switched to `exceptions-safe` here (which is indeed nice, also my comment isn't related to exactly this PR, but still), looks like migration was done not fully correctly. 

For instance, 
https://github.com/input-output-hk/cardano-sl/blob/d9e2fd91bc9e1a7b8a3d933c55c5a5f37c32c6dd/networking/src/NTP/Client.hs#L278
assumed to be cleanup action, and shouldn't be interrupted by async exceptions